src/test/resources/com/puppycrawl/tools/checkstyle/grammar/java19/ExpectedRecordPatternsPreviewNestedDecomposition.txt

```diff
@@ -0,0 +1,684 @@
+        |       |   |           |--LPAREN -> ( [38:34]
+        |       |   |           |--RECORD_PATTERN_COMPONENTS -> RECORD_PATTERN_COMPONENTS [38:35]
+        |       |   |           |   |--PATTERN_VARIABLE_DEF -> PATTERN_VARIABLE_DEF [38:35]
+        |       |   |           |   |   |--MODIFIERS -> MODIFIERS [38:35]
+        |       |   |           |   |   |--TYPE -> TYPE [38:35]
+        |       |   |           |   |   |   `--IDENT -> ColoredPoint [38:35]
+        |       |   |           |   |   `--IDENT -> ul [38:48]
+        |       |   |           |   |--COMMA -> , [38:50]
+        |       |   |           |   `--PATTERN_VARIABLE_DEF -> PATTERN_VARIABLE_DEF [38:51]
+        |       |   |           |       |--MODIFIERS -> MODIFIERS [38:51]
+        |       |   |           |       |--TYPE -> TYPE [38:51]
+        |       |   |           |       |   `--IDENT -> ColoredPoint [38:51]
+        |       |   |           |       `--IDENT -> lr [38:64]
+        |       |   |           `--RPAREN -> ) [38:66]
+        |       |   |--RPAREN -> ) [38:67]
+        |       |   `--SLIST -> { [38:69]
+        |       |       |--EXPR -> EXPR [39:30]
+        |       |       |   `--METHOD_CALL -> ( [39:30]
+        |       |       |       |--DOT -> . [39:22]
+        |       |       |       |   |--DOT -> . [39:18]
+        |       |       |       |   |   |--IDENT -> System [39:12]
+        |       |       |       |   |   `--IDENT -> out [39:19]
+        |       |       |       |   `--IDENT -> println [39:23]
+        |       |       |       |--ELIST -> ELIST [39:35]
+        |       |       |       |   `--EXPR -> EXPR [39:35]
+        |       |       |       |       `--METHOD_CALL -> ( [39:35]
+        |       |       |       |           |--DOT -> . [39:33]
+        |       |       |       |           |   |--IDENT -> ul [39:31]
+        |       |       |       |           |   `--IDENT -> c [39:34]
+        |       |       |       |           |--ELIST -> ELIST [39:36]
+        |       |       |       |           `--RPAREN -> ) [39:36]
+        |       |       |       `--RPAREN -> ) [39:37]
+        |       |       |--SEMI -> ; [39:38]
+        |       |       `--RCURLY -> } [40:8]
+        |       `--RCURLY -> } [41:4]
+        |--METHOD_DEF -> METHOD_DEF [45:4]
+        |   |--MODIFIERS -> MODIFIERS [45:4]
+        |   |   `--LITERAL_STATIC -> static [45:4]
+        |   |--TYPE -> TYPE [45:11]
+        |   |   `--LITERAL_VOID -> void [45:11]
+        |   |--IDENT -> p2 [45:16]
+        |   |--LPAREN -> ( [45:18]
+        |   |--PARAMETERS -> PARAMETERS [45:19]
+        |   |   `--PARAMETER_DEF -> PARAMETER_DEF [45:19]
+        |   |       |--MODIFIERS -> MODIFIERS [45:19]
+        |   |       |--TYPE -> TYPE [45:19]
+        |   |       |   `--IDENT -> Rectangle [45:19]
+        |   |       `--IDENT -> r [45:29]
+        |   |--RPAREN -> ) [45:30]
+        |   `--SLIST -> { [45:32]
+        |       |--LITERAL_IF -> if [46:8]
```

Comment:
This is the beginning of a complicated case of record pattern decomposition:

```
        if (r instanceof Rectangle(ColoredPoint(Point p1,Color c1),
                ColoredPoint lr1)
                && r instanceof Rectangle(ColoredPoint(Point p2,Color c2),
                ColoredPoint lr2) && lr2.c == Color.BLUE) {
            System.out.println(r);
        }





```

Reply:
it is easier to read at https://github.com/checkstyle/checkstyle/wiki/Record-Pattern-Preview---AST-strucuture